### DISCUSSION TITLE
Changing subdomains based on element types

### DISCUSSION THREAD

**Discussion by TheGreatCid** [idaholab/moose#25425] [Q&A General] [answered]
Hi all,

Does MOOSE have the capability to read a mesh and reassign the subdomains for elements of a specified type? 
For example, I have a mesh of HEX8's and TET4's on subdomain 1. I want to create a new subdomain, remove the HEX8's from subdomain 1, and move them onto the new subdomain.

Then, as an aside, if anyone knows how to perform this in GMSH, advice there would be welcome as well. 

Thank you!

**Comment by GiudGiud** [comment, marked as answer]
Yes

please use the meshrepairgenerator with the separate element type parameter 

you ll have to use this Generator after loading the mesh

**Reply by GiudGiud** [reply]
This is a fairly new capability you will need moose to be up to date for this

**Reply by TheGreatCid** [reply]
Just updated and tested it. It works great! Thank you for the response
